site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,323 @@
+| *correlation-id* | yes       | *properties*             | *message-id*    | Contains the *message-id* (or the *correlation-id*, if specified) of the request message that this message is the response to. |
+| *device_id*      | yes       | *application-properties* | UTF-8 *string*  | Contains the ID of the device. |
+| *tenant_id*      | yes       | *application-properties* | UTF-8 *string*  | Contains the ID of the tenant to which the device belongs. |
+| *status*         | yes       | *application-properties* | *int*           | Contains the status code indicating the outcome of the operation. Concrete values and their semantics are defined for each particular operation. |
+
+# Delivery States
+
+Hono uses the following AMQP message delivery states when receiving request messages from clients:
+
+| Delivery State | Description |
+| :------------- | :---------- |
+| *ACCEPTED*     | Indicates that Hono has successfully received and accepted the request for processing. |
+| *REJECTED*     | Indicates that Hono has received the request but was not able to process it. The *error* field contains information regarding the reason why. Clients should not try to re-send the request using the same message properties in this case. |
+
+# Payload Format
+
+Most of the operations of the Credentials API allow or require the inclusion of payload data as part of the
+request or response messages of the operation. Such payload is carried in the body of the corresponding AMQP 
+messages as part of a single *AMQP Value* section.
+
+## Request Payload
+
+The payload included in *request* messages consists of a UTF-8 encoded string representation of a single JSON object. It is an error to include payload that is not of this type.
+
+The JSON object MUST contain at least the following members:
+
+| Name             | Mandatory | Type       | Description |
+| :--------------- | :-------: | :--------- | :---------- |
+| *device-id*      | *yes*     | *string*   | The ID of the device to which the credentials belong. |
+| *type*           | *yes*     | *string*   | The credential type name. The value may be arbitrarily chosen by clients but SHOULD reflect the particular type of authentication mechanism the credentials are to be used with. Possible values include (but are not limited to) `psk`, `RawPublicKey`, `hashed-password` etc. |
+| *auth-id*        | *yes*     | *string*   | The identity that the device should be authenticated as. |
+
+For each set of credentials the combination of `auth-id` and `type` MUST be unique within a tenant.
+
+The JSON object MAY contain additional members of arbitrary other names which MUST be of a scalar type only.
+
+Below is an example for a payload of a request for adding [a hashed password]({{< ref "#hashed-password" >}}) for username `billie` using SHA256 as the hashing function with a 4 byte salt (Base64 encoding of `0x32AEF017`) to device `4711`:
```

Comment:
I would propose to add a constraint regarding the minimum lenght of the salt here.
Very short salts do not make cracking attacks much harder, and insisting on a minimum salt length does not cost much.
My proposal: we do not accept salts shorter than 4 bytes minimum. 
WDYT?